dashboard/app/controllers/transfers_controller.rb

```diff
@@ -0,0 +1,128 @@
+class TransfersController < ApplicationController
+  before_filter :authenticate_user!
+
+  # POST /section/:id/transfers
+  def create
+    new_section_code = params[:id]
+
+    begin
+      new_section = Section.find_by!(code: new_section_code)
+    rescue ActiveRecord::RecordNotFound
+      # TODO: i18n
+      render json: {
+        error: "Sorry, but that section does not exist. Please enter a different section code."
+      }, status: :not_found
+      return
+    end
+
+    # This is needed when we eventually allow students to be in multiple sections
+    if params.has_key?(:current_section_code)
+      current_section_code = params[:current_section_code]
+    else
+      # TODO: i18n
+      render json: {
+        error: "Please provide current_section_code."
+      }, status: :bad_request
+      return
+    end
+
+    begin
+      current_section = Section.find_by!(code: current_section_code)
+    rescue ActiveRecord::RecordNotFound
+      # TODO: i18n
+      render json: {
+        error: "Sorry, but section #{current_section_code} does not exist. Please enter a different section code."
+      }, status: :not_found
+      return
+    end
+
+    current_section_belongs_to_logged_in_user = current_user.sections.map(&:code).include?(current_section_code)
```

Comment:
This could be replaced with `new_section.user == current_user`. Given that this is clearer than the map/include code, I would probably refactor this to eliminate the temporary variable since it's only used once in the line below, so something like:
````
if new_section.user !-= current_user
      render json: {
        error: "You cannot move students from a section that does not belong to you."
      }, status: :forbidden
      return
    end
````